### PR TITLE
fix(elasticloadbalancingv2): `http2Enabled` with true is ignored in ApplicationLoadBalancer

### LINKED ISSUE #31609: ElasticLoadBalancingV2: ApplicationLoadBalancer http2.enabled from false to true is ignored in cloudformation
### Describe the bug

When an ApplicationLoadBalancer routing.http2.enabled attribute is modified from "false" to "true" CDK will remove the attribute with the expectation that "true" will be the default. CloudFormation does not remove the attribute nor does it set the attribute value back to default.

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Last Known Working CDK Version

_No response_

### Expected Behavior

Modifying the attribute value from "false" to "true" modifies the load balancer attribute from "false" to "true"

### Current Behavior

Value remains "false" despite the attribute being removed.

### Reproduction Steps

New instance of load balancer, http2.enabled set to false; deploy
Modify http2 from false to true; deploy

### Possible Solution

When http2 is not undefined, reflect string value

### Additional Information/Context

_No response_

### CDK CLI Version

2.160.0 (build 7a8ae02)

### Framework Version

_No response_

### Node.js Version

v18.19.1

### OS

Ubuntu 24.04.1 LTS

### Language

.NET

### Language Version

.Net 8.0.108

### Other information

_No response_

### PR DESCRIPTION
**Pull request by go-to-k**
### Issue # (if applicable)

Closes #31609.

### Reason for this change

The `http2Enabled` can only be set to CFn if it is specified with false. Therefore, if it is specified with true, the change is not reflected in CFn.

### Description of changes

Passed `http2Enabled` if it is not undefined.

```ts
if (props.http2Enabled !== undefined) { this.setAttribute('routing.http2.enabled', props.http2Enabled ? 'true' : 'false'); }
```

### Description of how you validated changes

Both unit and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/test/alb/load-balancer.test.ts [hunk 1]
```diff
@@ -264,6 +264,75 @@ describe('tests', () => {
...
+      Template.fromStack(stack).hasResourceProperties('AWS::ElasticLoadBalancingV2::LoadBalancer', {
+        LoadBalancerAttributes: Match.arrayWith([
+          {
+            Key: 'routing.http2.enabled',
+            Value: 'false',
+          },
+        ]),
+      });
+    });
+  });
```
Comment (badmintoncryer) [inline packages/aws-cdk-lib/aws-elasticloadbalancingv2/test/alb/load-balancer.test.ts:334]: How about integrate these 2 tests?

```suggestion
    test.each([true, false])('http2Enabled is set to %s', (http2Enabled) => {
      // GIVEN
      const stack = new cdk.Stack();
      const vpc = new ec2.Vpc(stack, 'Stack');

      // WHEN
      new elbv2.ApplicationLoadBalancer(stack, 'LB', {
        vpc,
        http2Enabled,
      });

      // THEN
      Template.fromStack(stack).hasResourceProperties('AWS::ElasticLoadBalancingV2::LoadBalancer', {
        LoadBalancerAttributes: Match.arrayWith([
          {
            Key: 'routing.http2.enabled',
            Value: String(http2Enabled),
          },
        ]),
      });
    });
```

**Review by badmintoncryer — APPROVED** [review]
Thanks!
I suggested a refactor comment. This modification is not mandatory, so please feel free to apply it if it suits your preference!

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/test/alb/load-balancer.test.ts (hunk 1 above)
Comment (go-to-k) [reply packages/aws-cdk-lib/aws-elasticloadbalancingv2/test/alb/load-balancer.test.ts]: Thanks! I have modified the tests.

**Review by moelasmar — APPROVED** [review]
It looks good to me

**Review by moelasmar — APPROVED** [review]

**Pull request merged**